### ISSUE TITLE
Bump `opensaml/opensaml-saml-impl` version

### ISSUE THREAD

**Issue by slipset** [metabase/saml20-clj#52]
`opensaml/opensaml-saml-impl` has a dependency on `velocity` version 1.7 which has  [CVE-2020-13936](https://nvd.nist.gov/vuln/detail/CVE-2020-13936) against it

**Comment by mrrodriguez** [comment]
I don't think this is a trivial change.
https://shibboleth.atlassian.net/wiki/spaces/OS30/overview

opensaml 3.x is past its End of Life for support. The fix for the `velocity` dependency requires going to the next major version of it, which has breaking API changes (and even artifact organization).

I think the only path forward is to move to opensaml 4.x

It isn't clear to me if this project has anything logged for that or what that involves.

**Comment by camsaul** [comment]
So unfortunately it looks like 4.x only works on Java 11+: 

https://shibboleth.atlassian.net/wiki/spaces/OSAML/overview

> OpenSAML 4, the current Java library version, is based on Java 11, and supports SAML 1.0, 1.1, and 2.0. Additionally, various development groups have found the framework created to support OpenSAML useful for their own work and the Java codebase includes some code supporting WS-Addressing, WS-Security, WS-Trust and XACML.
>
> The previous major version based on Java 8 is now End of Life and no longer supported.

With Metabase at least we support Java 8+. I'm reverting the version bump in #56 for now until I can figure out a way around this. I might end up releasing two separate artifacts, one with v3 targeting Java 8 and one with v4 targeting Java 11

**Comment by mrrodriguez** [comment]
@camsaul I found no way to go forward with opensaml due to the JVM 11  requirement. Also, reading their docs does not make it sound like a good choice in general.

They have warnings about using the lib outside of something like `shibboleth`, but also this is in the docs:

> We do not have significant documentation for this version of the library. Some of the concepts and examples in the (now quite old) [OpenSAML 2](https://shibboleth.atlassian.net/wiki/spaces/OpenSAML) Java documentation can be applied, with varying degrees of change, to this version (and that documentation is itself not the best). But you should recognize the inherent risk of relying on this library and you should not expect substantial improvement in this area.

To me it sounds like an open source component of `shibboleth` they do not plan to be transparent about for consumers. I also think the requirement for Java 11 is quite aggressive at this point. They give no migration details and also are not going to update their Java 8 supporting version to remove security vulnerabilities - such as the one from the `velocity` library.

Due to this, I've given up hope on continuing on being reliant on opensaml. So far I've found https://github.com/onelogin/java-saml to be a sufficient alternative. It looks like there is enough in it to accomplish the same level of features as what I see provided by this `saml02-clj` lib over opensaml. 
I think actually, the best case, would be for this project to move to `java-saml` entirely as well, however, easier said than done I'm sure.
